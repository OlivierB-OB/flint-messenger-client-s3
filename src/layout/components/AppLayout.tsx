import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import React from 'react';
import { connect } from 'react-redux';
import { AppDrawer, drawerWidth } from './AppDrawer';
import { AppMenu } from './AppMenu';
import { AppContent } from './AppContent';
import { IAppState } from '../../appReducer';

interface ILayoutDisplayProps {
  showDrawer: boolean;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      width: `100%`,
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    contentShift: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    drawer: {
      width: drawerWidth,
    },
  }),
);

export function LayoutDisplay({ showDrawer }: ILayoutDisplayProps) {
  const classes = useStyles();
  const contentClasses = [classes.content, showDrawer && classes.contentShift].filter(Boolean).join(' ');
  return (
    <div>
      <div className={contentClasses}>
        <AppMenu />
        <AppContent />
      </div>
      <AppDrawer />
    </div>
  );
}

const mapStateToProps = ({ layout }: IAppState) => ({
  showDrawer: layout.showDrawer,
});

export const AppLayout = connect(mapStateToProps)(LayoutDisplay);
