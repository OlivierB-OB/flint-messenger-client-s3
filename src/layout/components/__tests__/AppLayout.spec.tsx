import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { AppLayout } from '../AppLayout';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('AppLayout', () => {
  let Connected: any = AppLayout;

  it('should provide showDrawer to LayoutDisplay', async () => {
    const store = mockStore({
      layout: {
        showDrawer: 'showDrawer from state',
      },
    });
    const component = shallow(<Connected store={store} />).dive();
    expect(component).toMatchSnapshot();
  });
});
