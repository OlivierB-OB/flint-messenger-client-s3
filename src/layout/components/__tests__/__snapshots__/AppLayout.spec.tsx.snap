// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout should provide showDrawer to LayoutDisplay 1`] = `
<LayoutDisplay
  dispatch={[Function]}
  showDrawer="showDrawer from state"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
